src/routes/auth/data.ts

```diff
@@ -3,7 +3,15 @@ import type { Profile, SignInData } from './validate.ts';
 
 export async function getPassword(database: D1Database, body: SignInData): Promise<string> {
 	const { results } = await database
-		.prepare(`SELECT password FROM ${DBTables.PROFILE} WHERE email = ? AND activatedAt IS NOT NULL LIMIT 1`)
+		.prepare(`
+            SELECT password
+            FROM ${DBTables.ACCESS}
+            WHERE id IN (
+                SELECT id
+                FROM ${DBTables.PROFILE}
+                WHERE email = ? AND activatedAt IS NOT NULL
+            )
```

Comment:
I recommend using a JOIN query instead of a nested subquery. JOIN explicitly defines the relationship between tables, which enhances clarity and may optimize query performance. This approach can improve both readability and efficiency.
And,, don't forget to set LIMIT 1

```suggestion
            SELECT a.password
            FROM ${DBTables.PROFILE} p
            JOIN ${DBTables.ACCESS} a ON a.id = p.id
            WHERE p.email = ? AND p.activatedAt IS NOT NULL
            LIMIT 1
```
ref)
https://stackoverflow.com/questions/2577174/join-vs-sub-query
https://www.geeksforgeeks.org/sql-join-vs-subquery/

Reply:
Yeah, no worries, but my question is more like: why don't we keep all of the data in one table?

Like:
```sql
CREATE TABLE IF NOT EXISTS access (
	id TEXT NOT NULL UNIQUE COLLATE BINARY,
	schemaVersion INTEGER NOT NULL DEFAULT 1,
	access TEXT NOT NULL CHECK(access IN ('admin', 'organizer', 'volunteer')),
	password TEXT NOT NULL,

	email TEXT NOT NULL UNIQUE, -- <= Add it here
	activatedAt DATETIME DEFAULT NULL, -- <= And here

	PRIMARY KEY (id),
	FOREIGN KEY (id) REFERENCES profile(id)
);
```

So the query would only look for one table instead of multiple. It _does_ duplicate information, but may be interesting for performance.

The argument here is as this is the auth endpoint and a verification we will need to do multiple times, potentially on every request (as part of the auth middleware), we would benefit from a _very fast_ query.

Reply:
Sounds reasonable! yeah, there'll be a lot of requests to get password, so we would benefit from a fast query.

Reply:
According to the geeks for geeks article utilizing a join is bad for readability but is better for performance.   I opted for sub query for readability and assumed performance wouldn't be an issue due to the volume expected from the platform.